### DISCUSSION TITLE
Is there a way to only check for force_input while moving foreward in an experiment but not while moving backwards?

### DISCUSSION THREAD

**Discussion by Nudelkoenig** [ctreffe/alfred#110] [Alfred Q&A] [answered]
When using sections that allow backwards movement in combination with "force_input"-elements one has to fill in every "force-input"-element in order to be allowed to actually move backwards. Is there a way to change this behaviour, so that moving backwards does not require to first fill in every "force_input"-element on the page one is currently on but moving forward does?

**Comment by jobrachem** [comment, marked as answer]
You should be able to achieve the desired result by subclassing `Section` and overloading the `validate_on_backward` method:

```python
import alfred3 as al

class FreeBackwardSection(al.Section):
    def validate_on_backward(self):
        pass
```

This will deactivate all validation on backwards moves in that section. 

You can also implement custom validation if you wish. In this case, you need to raise `alfred3.exceptions.ValidationError` in case validation fails. For instance, the following code will re-implement the default behavior:

```python
import alfred3 as al
from alfred3.exceptions import ValidationError

class FreeBackwardSection(al.Section):

    def validate_on_backward(self):
        if not self.exp.current_page._validate_elements():
            raise ValidationError()
        
        if not self.exp.current_page._validate():
            raise ValidationError()
```
